Servlets/src/main/java/ua/goit/servlets/servlets/ProjectsServlet.java

```diff
@@ -0,0 +1,56 @@
+package ua.goit.servlets.servlets;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import ua.goit.servlets.dao.CategoriesList;
+import ua.goit.servlets.model.Category;
+import ua.goit.servlets.model.Project;
+
+public class ProjectsServlet extends HttpServlet {
+    static final long serialVersionUID = 15909L;
+    List<Category> categories; //visible for testing
+
+    public void init() throws ServletException {
+	super.init();
+	categories = CategoriesList.getAllCategories();
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response)throws IOException, ServletException {
+	String categoryIDstring = request.getParameter("category");
+	int categoryID = Integer.valueOf(categoryIDstring);
+
+	List<Project> projects = getProjectsForCategory(categoryID);
+	writeProjectsList(projects, response);
+    }
+
+    private List<Project> getProjectsForCategory(int categoryID) {
```

Comment:
I remember asking a question if this can be made faster than just iterating over all available projects